### ISSUE TITLE
keep remaining agents running

### ISSUE THREAD

**Issue by Wouter1** [matrx-software/matrx#239]
**Describe the bug**
It seems that a single crashed agent blocks this whole system. This is also not what I expect. In the real world agents may crash all the time, it's up to the remaining agents to fix the problems.

**To Reproduce**
have a team and let one agent crash.
The whole system halts. 

**Expected behavior**
In a cooperative agent system, I think remaining agents should take over the remaining tasks, or even spawn a replacement agent if possible.

**Comment by jwaa** [comment]
@Wouter1 thank you for submitting the issue. We are discussing it at the moment.

**Comment by jwaa** [comment]
@Wouter1 after some internal discussion we decided that this is not a bug. 

If do not raise an exception from within `AgentBrain` we will encounter several difficulties. Such as that during an experiment the participant can just continue with the task, while potentially vital functionality of the experimental task failed. Furthermore, to some extent, debugging becomes more less transparent if exceptions are not raised. If they are printed or logged as a `Warning` instead this will clutter the console/log output. 

As such we decided to close this issue.

If you do want to prevent agents from stopping MATRX on an exception, there are several options available. An easy, reusable and transparent one is through decorators. For example the a decorator factory such as this:

```python
def prevent_exceptions(*exception_classes):
    def decorator(func):
        def wrapper(*args, **kwargs):
            try:
                func(*args, **kwargs)
            except Exception as e:
                if not isinstance(e, *exception_classes):
                    raise e
        return wrapper
    return decorator

@prevent_exceptions(UserWarning, IndexError)
def does_not_raise_indexerror_and_userwarning(...): ...
```

It would catch all exceptions of a given class and not raise it, while raising all other exceptions. Note that this also allows ignoring all exceptions which is generally not advised (see [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)).
